diff --git a/arch-update/arch-update b/arch-update/arch-update
index 89f244c..587a695 100755
--- a/arch-update/arch-update
+++ b/arch-update/arch-update
@@ -66,6 +66,14 @@ def create_argparse():
         help='Explicitly watch for specified packages. '
         'Listed elements are treated as regular expressions for matching.'
     )
+    parser.add_argument(
+        '-l',
+        '--label-only',
+        action = 'store_const',
+        const = True,
+        default = _default('LABEL_ONLY', 'False', strbool),
+        help = 'Show only label if updates available.'
+    )
     return parser.parse_args()
 
 
@@ -137,8 +145,12 @@ elif args.aur_yay:
     updates += get_aur_yay_updates()
 
 update_count = len(updates)
+
 if update_count > 0:
-    if update_count == 1:
+    if args.label_only:
+      info = ''
+      short_info = ''
+    elif update_count == 1:
       info = str(update_count) + ' update available'
       short_info = str(update_count) + ' update'
     else:
diff --git a/arch-update/arch-update.sh b/arch-update/arch-update.sh
new file mode 100755
index 0000000..86b893f
--- /dev/null
+++ b/arch-update/arch-update.sh
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+if [[ $1 = "update" ]]; then
+    echo "Executing yay -Syu"
+    # NOTE: Could call some eos update script
+    yay -Syu
+    echo "Upgrade finished, you may close the window."
+    exit 0
+fi
+
+if [[ $BLOCK_BUTTON -eq 3 ]]; then
+    $TERMINAL -H -x ~/.config/i3/i3blocks-contrib/arch-update/arch-update.sh update
+fi
+
+~/.config/i3/i3blocks-contrib/arch-update/arch-update --label-only
+
diff --git a/bandwidth2/bandwidth2.c b/bandwidth2/bandwidth2.c
index d354b4b..cfae3fc 100644
--- a/bandwidth2/bandwidth2.c
+++ b/bandwidth2/bandwidth2.c
@@ -94,20 +94,22 @@ void display(int const unit, int const divisor,
   if (unit == 'b')
     b = b * 8;
 
-  snprintf(fmtstr, sizeof (fmtstr), "%%%d.1lf", divisor > 1000 ? 6 : 5);
+  // NOTE: changed precision to 0 decimals and length - 2
+  snprintf(fmtstr, sizeof (fmtstr), "%%%d.0lf", divisor > 1000 ? 3 : 2);
 
+  // NOTE: removed /s
   if (b < divisor) {
     printf(fmtstr, b);
-    printf(" %c/s", unit);
+    printf(" %c", unit);
   } else if (b < divisor * divisor) {
     printf(fmtstr, b / divisor);
-    printf("K%c/s", unit);
+    printf("K%c", unit);
   } else if (b < divisor * divisor * divisor) {
     printf(fmtstr, b / (divisor * divisor));
-    printf("M%c/s", unit);
+    printf("M%c", unit);
   } else {
     printf(fmtstr, b / (divisor * divisor * divisor));
-    printf("G%c/s", unit);
+    printf("G%c", unit);
   }
   printf("</span>");
 }
@@ -213,9 +215,9 @@ int main(int argc, char *argv[])
     tx = (sent - sent_old) / (float)(s - s_old);
     printf("%s", label);
     display(unit, divisor, rx, warningrx, criticalrx);
-    printf(" ");
+    printf(" ");
     display(unit, divisor, tx, warningtx, criticaltx);
-    printf("\n");
+    printf("\n");
     fflush(stdout);
     s_old = s;
     received_old = received;
diff --git a/gpu-load/gpu-load b/gpu-load/gpu-load
index 4966219..238cd6a 100755
--- a/gpu-load/gpu-load
+++ b/gpu-load/gpu-load
@@ -67,7 +67,7 @@ elsif ($gpu_brand eq "AMD") {
         }
     }
     close(AMD);
-    $full_text = sprintf "%.0f%% %.0f%%\n", $gpu_usage, $gpu_mem;
+    $full_text = sprintf " %.0f%%   %.0f%%\n", $gpu_usage, $gpu_mem;
 }
 
 $gpu_usage eq -1 and die 'Can\'t find GPU information';
diff --git a/mediaplayer/mp-scrolling b/mediaplayer/mp-scrolling
new file mode 100755
index 0000000..1f88c60
--- /dev/null
+++ b/mediaplayer/mp-scrolling
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+STR=$(~/.config/i3/i3blocks-contrib/mediaplayer/mediaplayer)
+LENGTH=20
+SYSTEM_TIME=$(date +%s)
+INDEX=$(( 1 + ($SYSTEM_TIME % $LENGTH) * 2 ))
+
+#echo "$INDEX"
+if [[ ! -z "$STR" ]]; then
+    # BUG: Cut might corrupt a UTF8-character if it happens to cut between the utf8-bytes
+    echo "$STR | $STR | $STR" | cut "-c$INDEX-$(($INDEX + $LENGTH))"
+fi
+
diff --git a/mediaplayer/mp-switching b/mediaplayer/mp-switching
new file mode 100755
index 0000000..176439f
--- /dev/null
+++ b/mediaplayer/mp-switching
@@ -0,0 +1,26 @@
+#!/usr/bin/bash
+a=0
+
+# src: Cattendeavour miei scripts
+
+path=~/.config/i3/i3blocks-contrib/mediaplayer/mp-switching
+
+if [ $a == 0 ]; then
+  sed -i '2d' $path 
+  sed -i '2ia=1' $path
+  # se non mostri la variabile subito ci sarà il delay di interval
+  a=1 
+elif [ $a == 1 ]; then
+  sed -i '2d' $path
+  sed -i '2ia=0' $path
+  a=0
+fi
+
+
+STR=$(~/.config/i3/i3blocks-contrib/mediaplayer/mediaplayer)
+IFS='-' read -ra STR <<< "$STR"
+
+if [[ ! -z "$STR" ]]; then
+  echo ${STR[$a]} | cut -c1-20
+fi
+
