diff --git a/bandwidth2/bandwidth2.c b/bandwidth2/bandwidth2.c
index d354b4b..43c3984 100644
--- a/bandwidth2/bandwidth2.c
+++ b/bandwidth2/bandwidth2.c
@@ -94,7 +94,8 @@ void display(int const unit, int const divisor,
   if (unit == 'b')
     b = b * 8;
 
-  snprintf(fmtstr, sizeof (fmtstr), "%%%d.1lf", divisor > 1000 ? 6 : 5);
+  // NOTE: changed precision to 0 decimals and length - 2
+  snprintf(fmtstr, sizeof (fmtstr), "%%%d.0lf", divisor > 1000 ? 3 : 2);
 
   if (b < divisor) {
     printf(fmtstr, b);
@@ -211,11 +212,11 @@ int main(int argc, char *argv[])
 
     rx = (received - received_old) / (float)(s - s_old);
     tx = (sent - sent_old) / (float)(s - s_old);
-    printf("%s", label);
+    printf("%s", label);
     display(unit, divisor, rx, warningrx, criticalrx);
-    printf(" ");
+    //printf("");
     display(unit, divisor, tx, warningtx, criticaltx);
-    printf("\n");
+    printf(" \n");
     fflush(stdout);
     s_old = s;
     received_old = received;
diff --git a/gpu-load/gpu-load b/gpu-load/gpu-load
index 4966219..238cd6a 100755
--- a/gpu-load/gpu-load
+++ b/gpu-load/gpu-load
@@ -67,7 +67,7 @@ elsif ($gpu_brand eq "AMD") {
         }
     }
     close(AMD);
-    $full_text = sprintf "%.0f%% %.0f%%\n", $gpu_usage, $gpu_mem;
+    $full_text = sprintf " %.0f%%   %.0f%%\n", $gpu_usage, $gpu_mem;
 }
 
 $gpu_usage eq -1 and die 'Can\'t find GPU information';
